plugins/modules/pacemaker_stonith.py

```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2025, Dexter Le <dextersydney2001@gmail.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: pacemaker_stonith
+short_description: Manage pacemaker STONITH
+author:
+  - Dexter Le (@munchtoast)
+version_added: 11.1.0
+description:
+  - This module can manage STONITH in a Pacemaker cluster using the pacemaker CLI.
+extends_documentation_fragment:
+  - community.general.attributes
+attributes:
+  check_mode:
+    support: full
+  diff_mode:
+    support: none
+options:
+  state:
+    description:
+      - Indicate desired state for cluster STONITH.
+    choices: [ present, absent, enabled, disabled ]
+    default: present
+    type: str
+  name:
+    description:
+      - Specify the STONITH name to create.
+    required: true
+    type: str
+  stonith_type:
+    description:
+      - Specify the STONITH device type
+    type: str
+  stonith_options:
+    description:
+      - Specify the STONITH option to create.
+    type: list
+    elements: str
+    default: []
+  stonith_operations:
+    description:
+      - List of operations to associate with STONITH.
+    type: list
+    elements: dict
+    default: []
+    suboptions:
+      operation_action:
+        description:
+          - Operation action to associate with STONITH.
+        type: str
+      operation_options:
+        description:
+          - Operation option to associate with action.
+        type: list
+        elements: str
+  stonith_metas:
+    description:
+      - List of meta to associate with STONITH.
+    type: list
+    elements: str
+  stonith_argument:
+    description:
+      - Action to associate with STONITH.
+    type: dict
+    suboptions:
+      argument_action:
+        description:
+          - Action to apply to STONITH.
+        type: str
+        choices: [ group, before, after ]
+      argument_options:
+        description:
+          - Options to associate with STONITH action.
+        type: list
+        elements: str
+  agent_validation:
+    description:
+      - enabled agent validation for STONITH creation.
+    type: bool
+    default: false
+  wait:
+    description:
+      - Timeout period for polling the STONITH creation.
+    type: int
+    default: 300
+'''
+
+EXAMPLES = '''
+---
+- name: Create pacemaker STONITH
+  hosts: localhost
+  gather_facts: false
+  tasks:
+  - name: Create virtual-ip STONITH
+    community.general.pacemaker_stonith:
+      state: present
+      name: virtual-stonith
+      stonith_type: fence_virt
+      stonith_options:
+        - "pcmk_host_list=f1"
+      stonith_operations:
+        - operation_action: monitor
+          operation_options:
+            - "interval=30s"
```

Comment:
Usually module examples are tasks, not playbooks. Since this playbook has a single task and the playbook part is pretty much boilderplate and unrelated to this example, how about removing it?
```suggestion
- name: Create virtual-ip STONITH
  community.general.pacemaker_stonith:
    state: present
    name: virtual-stonith
    stonith_type: fence_virt
    stonith_options:
      - "pcmk_host_list=f1"
    stonith_operations:
      - operation_action: monitor
        operation_options:
          - "interval=30s"
```